modules/dxp/apps/portal-search-similar-results/portal-search-similar-results-web/src/main/java/com/liferay/portal/search/similar/results/web/internal/contributor/asset/publisher/AssetPublisherSimilarResultsContributor.java

```diff
@@ -139,23 +164,53 @@ private Long _getId(AssetEntry assetEntry) {
 				return null;
 			}
 
-			return journalArticle.getId();
+			return journalArticle;
+		}
+		else if (Objects.equals(
+					WikiPage.class.getName(), assetEntry.getClassName())) {
+
+			WikiPage wikiPage =
+				_wikiPageLocalService.fetchWikiPageByUuidAndGroupId(
+					assetEntry.getClassUuid(), assetEntry.getGroupId());
+
+			if (wikiPage == null) {
+				return null;
+			}
+
+			return wikiPage;
+		}
```

Comment:
What are your thoughts on simplifying this? The `return null` could make it clear that a `null` value could be returned, but it could be redundant.

```java
else if (Objects.equals(
			WikiPage.class.getName(), assetEntry.getClassName())) {

	return _wikiPageLocalService.fetchWikiPageByUuidAndGroupId(
		assetEntry.getClassUuid(), assetEntry.getGroupId());
}
```

If so, we can also make this change for the other `if` checks.

Reply:
nice! make changes please.....